registry/remote/credentials/memory_store.go

```diff
@@ -17,23 +17,50 @@ package credentials
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"sync"
 
 	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/credentials/internal/config"
 )
 
-// memoryStore is a store that keeps credentials in memory.
-type memoryStore struct {
+// MemoryStore is a store that keeps credentials in memory.
+type MemoryStore struct {
```

Comment:
@Wwwsylvia What do you think about exposing the MemoryStore?

Reply:
I don't think exposing the type is necessary at this moment.